gomod-sync/cmd/config/version.go

```diff
@@ -0,0 +1,60 @@
+package config
+
+import "sort"
+
+// VersionConfig represents the version configuration
+type VersionConfig struct {
+	// Default version for the go.mod files
+	Default string `json:"default"`
+	// List of exercises and their versions that must not have
+	// the default version.
+	Exceptions []ExerciseVersion `json:"exceptions"`
+}
+
+// ExerciseExpectedVersion gives the expected version the go.mod file for
+// an exercise should be at. The argument should be the slug of an exercise.
+func (vc *VersionConfig) ExerciseExpectedVersion(exercise string) string {
+	for _, exception := range vc.Exceptions {
+		if exception.Exercise == exercise {
+			return exception.Version
+		}
+	}
+	return vc.Default
+}
+
+// Equal tells if this VersionConfig is equal to another VersionConfig
+func (vc *VersionConfig) Equal(other VersionConfig) bool {
+	return vc.Default == other.Default && equalExceptions(vc.Exceptions, other.Exceptions)
+}
+
+// equalExceptions compares two lists of exercise versions and tells if they are equal.
+// Two exercise list versions are considered equal if they contain the same elements,
+// regardless of their order in both lists.
+func equalExceptions(a, b []ExerciseVersion) bool {
```

Comment:
We don't strictly need it. At the time I thought it was a good idea to make this equal function not care about the order of the elements, but I understand it might complicate things unnecessarily. I removed that logic - the elements now must be in the same order in the JSON and in the config struct.